.github/scripts/check-pr.js

```diff
@@ -5,73 +5,274 @@ const { GITHUB_REPOSITORY, GITHUB_PR_NUMBER } = process.env;
 
 const [owner, repo] = GITHUB_REPOSITORY.split('/');
 
-async function getPRDetails() {
-    const url = `https://api.github.com/repos/${owner}/${repo}/pulls/${GITHUB_PR_NUMBER}`;
-    const response = await axios.get(url, {
-        headers: {
-            Authorization: `token ${githubToken}`
+async function getPRDetails(prNumber) {
+    const url = `https://api.github.com/repos/${owner}/${repo}/pulls/${prNumber}`;
+    try {
+        const response = await axios.get(url, {
+            headers: {
+                Authorization: `token ${githubToken}`,
```

Comment:
Which workflow executes this script and what are the effective permissions of the `const githubToken = process.env.GITHUB_TOKEN;` token?

Reply:
This is run as part of the `pr-label-milestone-check.yml` workflow.  The GITHUB_TOKEN is pulled from the secrets and assigned in that workflow:
```
     - name: Check PR labels and milestones
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          GITHUB_PR_NUMBER: ${{ github.event.number }}
        run: node .github/scripts/check-pr.js

Reply:
@ebadiere Ok so the effective permissions are `read` for all by virtue of the `pr-label-milestone-check.yml` workflow definition.